include/tcnn/TextUtils.hpp

```diff
@@ -0,0 +1,71 @@
+///
+/// @par Text / character utilities
+///
+
+#ifndef TCNN_TEXT_UTILS_H
+#define TCNN_TEXT_UTILS_H
+
+#include <cstddef>
+#include <cstdint>
+#include <string_view>
+
+#include <strong_type/equality.hpp>
+#include <strong_type/ordered.hpp>
+#include <strong_type/strong_type.hpp>
+
+namespace tcnn {
+
+using image_id = strong::type<std::int8_t, struct image_id_tag, strong::ordered,
+                              strong::equality>;
+
+inline constexpr image_id TEXT_PAD_ID{0};
+inline constexpr image_id TEXT_VOCAB_SIZE{96}; // Number of printable ASCII
+                                               // characters (from space to ~)
+inline constexpr image_id TEXT_FIRST_PRINTABLE_CHAR{32};
+inline constexpr image_id TEXT_LAST_PRINTABLE_CHAR{126};
+inline constexpr image_id TEXT_PRINTABLE_RANGE_OFFSET{-31};
+
+struct TextGridParams {
+  std::string_view text;
+  std::size_t width{0};
+  std::size_t maxRows{0};
+};
+
+///
+/// @brief Checks if a character code is a printable ASCII character.
+/// @param CODE The character code to check.
+/// @return true if the character is printable, false otherwise.
+///
+[[nodiscard]] constexpr auto is_printable_char(const std::uint8_t CODE) noexcept
+    -> bool {
+  return (image_id{CODE} >= TEXT_FIRST_PRINTABLE_CHAR &&
+          image_id{CODE} <= TEXT_LAST_PRINTABLE_CHAR);
+}
+
+///
+/// @brief Converts a character to its corresponding ID in the range [1,
+/// TEXT_CONSTANTS_VOCAB_SIZE]. The ID 0 is reserved for the padding character.
+///
+/// @param c The character to convert.
+/// @return The corresponding ID of the character.
+///
+[[nodiscard]] constexpr auto char_to_id(const char CHARACTER) noexcept
```

Comment:
[[nodiscard]] is used by the compiler to error out anyone ignoring the return value.  Since these functions are pure, ignoring the return value is almost certainly a bug.  We want to guard against that at compile time :)